lib/fire_boll_game.dart

```diff
@@ -54,19 +63,31 @@ class FireBollGame extends Game {
     enemies.removeWhere((enemy) {
       bool leaveScreen = enemy.position.top >= screenSize.height;
       if (leaveScreen) {
-        this.onLostGame();
+        lostGame = true;
       }
       return leaveScreen || enemy.inPlayer;
     });
   }
 
   @override
   void render(Canvas canvas) {
-    player.render(canvas, playerColor);
+    if (lostGame) {
```

Comment:
Render method is for render only, add logic of the game on the update method

Reply:
Moved to update method